packages/flutter/lib/src/gestures/long_press.dart

```diff
@@ -19,20 +22,36 @@ class LongPressGestureRecognizer extends PrimaryPointerGestureRecognizer {
   /// Consider assigning the [onLongPress] callback after creating this object.
   LongPressGestureRecognizer({ Object debugOwner }) : super(deadline: kLongPressTimeout, debugOwner: debugOwner);
 
+  //is a flag which is used to determine if the longPress action happened
+  bool _longPressAccepted = false;
+
   /// Called when a long-press is recognized.
   GestureLongPressCallback onLongPress;
+  
+  /// Called when long-press is recognized and the pointer stops to contact the screen
+  GestureLongPressUpCallback onLongPressUp;
 
   @override
   void didExceedDeadline() {
     resolve(GestureDisposition.accepted);
+    _longPressAccepted = true;
     if (onLongPress != null)
       invokeCallback<void>('onLongPress', onLongPress);
   }
 
   @override
   void handlePrimaryPointer(PointerEvent event) {
-    if (event is PointerUpEvent)
-      resolve(GestureDisposition.rejected);
+    if (event is PointerUpEvent) {
+      /// only accept the event if the pointer is for a longer period of a time on the screen
+      /// and the onLongPressUp event is registered
+      if( _longPressAccepted == true && onLongPressUp != null ) {
+        _longPressAccepted= false;
+        resolve(GestureDisposition.accepted);
```

Comment:
You don't need to resolve it again; if you reach this point then you know you've already accepted it.